apps/admin/frontend/src/components/deprecated_print_button.tsx

```diff
@@ -0,0 +1,119 @@
+import React, { useContext, useState, useCallback } from 'react';
+
+import {
+  Button,
+  StyledButtonProps,
+  Modal,
+  useCancelablePromise,
+  P,
+} from '@votingworks/ui';
+import { sleep } from '@votingworks/basics';
+import { Loading } from './loading';
+import { AppContext } from '../contexts/app_context';
+
+const DEFAULT_PROGRESS_MODAL_DELAY_SECONDS = 3;
+
+interface PrinterNotConnectedModalProps {
+  isNotConnected: boolean;
+  onContinue: () => void;
+  onClose: () => void;
+}
+
+function PrinterNotConnectedModal({
+  isNotConnected,
+  onContinue,
+  onClose,
+}: PrinterNotConnectedModalProps): JSX.Element {
+  return (
+    <Modal
+      title={
+        isNotConnected
+          ? 'The printer is not connected.'
+          : 'The printer is now connected.'
+      }
+      content={
+        isNotConnected ? (
+          <P>Please connect the printer.</P>
+        ) : (
+          <P>You may continue printing.</P>
+        )
+      }
+      actions={
+        <React.Fragment>
+          <Button
+            variant="primary"
+            onPress={onContinue}
+            disabled={isNotConnected}
+          >
+            Continue
+          </Button>
+          <Button onPress={onClose}>Close</Button>
+        </React.Fragment>
+      }
+    />
+  );
+}
+
+interface PrintButtonProps extends StyledButtonProps {
+  print: () => Promise<void>;
+  useDefaultProgressModal?: boolean;
+}
+
+export function DeprecatedPrintButton({
```

Comment:
There's also the `@deprecated` JSDoc tag that shows up as a strikethrough in VS Code. Also, for deprecated things I like to add a comment that explains when and how it should be removed. Either an explanation inline or a link to a Github issue.

Reply:
I don't intend for this deprecated component to ever be merged to main, this is mostly for me.